package.json

```diff
@@ -46,6 +46,7 @@
   },
   "resolutions": {
     "**/@graphql-codegen/cli/**/ws": "^7.4.6",
+    "graphql-config": "^4.1.0",
```

Comment:
Ran into a weird type def error (when running `tsc:full`) which I'm not too sure what's happening but I suspect `yarn` is doing something weird with resolving versions correctly. What I gathered is that updating `graphiql` bumps `codemirror-graphql` which bumps `graphql-language-service` and in the latest version, `graphql-config` was [removed](https://github.com/backstage/backstage/pull/11089/files#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2deL13588) as a direct dep ([ref](https://github.com/graphql/graphiql/commit/c36504a804d8cc54a5136340152999b4a1a2c69f#diff-3c54a63be7298ecf874f61083633f9e943a5f606cdd5268e0c60520a4cdcd7b1L34)). 

This results in `graphql-config` v3.4.1 to be installed at the root of `node_modules` (in `master`, v4.1.0 is installed at root). The problem arises from v3.4.1 (and nested deps from this version) depending on `graphql` v15 but that is listed as a peer dep so it doesn't get installed in the nested `node_modules` of the package resulting in resolving `graphql` from root which is v16 and has mismatched type defs. I wasn't able to force yarn to install a nested v15 of `graphql` in the `graphql-config` v3.4.1 package since yarn resolutions doesn't seem to support resolving peer deps. 

FYI, `graphql-config` v3.4.1 comes from [`eslint-plugin-graphql`](https://www.npmjs.com/package/eslint-plugin-graphql) (from `plugin-catalog-graphql`) and hasn't been updated in 2 years. `graphql-config` v4 is being requested by everything else in the project. I'm not too sure why this issue doesn't occur in master since the project only installs `graphql` v16 but I suspect is has to do with how `tsc` [`libCheck`](https://www.typescriptlang.org/tsconfig#skipLibCheck) works which states `"TypeScript will type check the code you specifically refer to in your app’s source code."` which may also mean it only checks against package typedefs at the root of your `node_modules` since the only difference is that in this branch, `graphql-config` v3.4.1 was pulled to the root instead of v4.

Alternatively, I tried downgrading the `graphql` dep in `plugin-catalog-graphql` to v15 to satisfy the peer dep which passes `tsc` but breaks at runtime with the error `Duplicate "graphql" modules cannot be used at the same time since different versions may have different capabilities and behavior` so this won't work unless we ensure every package in this project uses v15 instead.

So ... adding a resolution to force using v4 of `graphql-config` to make everything happy seems to be the best option (again this should only effect `plugin-catalog-graphql` who's requiring `eslint-plugin-graphql` as a devDep) 

sorry for my rambling of an explanation of my investigation but let me know if this makes sense or if I completely misdiagnosed the issue here 😅

Reply:
![confused](https://media.giphy.com/media/3oz8xZvvOZRmKay4xy/giphy.gif)

Reply:
Fantastic work troubleshooting this issue @kuangp and thanks for all the context! 🏅 

Given the state of graphql in Backstage I wonder if it make sense dropping the abandoned [eslint-plugin-graphql](https://www.npmjs.com/package/eslint-plugin-graphql) to prevent the outdated dependency from being pulled in at all. That would also remove the need for the resolution right?

other @backstage/maintainers WDYT?

Reply:
@kuangp I checked with the others and let's go for removing [eslint-plugin-graphql](https://www.npmjs.com/package/eslint-plugin-graphql) and see if that fixes it(no need for the resolution) 😁

Reply:
Diagnose is probably on point :grin: I think we can fix it by removing the `eslint-plugin-graphql` though, doesn't seem like it's actually used anyway. Resolutions is something we only want to use if absolutely necessary, since it's something that needs to be manually applied to all apps out there.